components/table/Column.razor.cs

```diff
@@ -114,7 +114,11 @@ public TData Field
 
         public LambdaExpression GetFieldExpression { get; private set; }
 
-        void IFieldColumn.ClearSorter() => SetSorter(SortDirection.None);
+        void IFieldColumn.ClearSorter()
+        {
+            SetSorter(SortDirection.None);
+            StateHasChanged();
```

Comment:
Does this problem only occur when using DataIndex? If so, please add a judgment condition.